<?xml version="1.0"?>
<launch>
    <arg name="marker_size"                 default="0.100"     doc="Size of the ArUco marker used, in meters" />
    <arg name="marker_id"                   default="566"       doc="The ID of the ArUco marker used" />
    <arg name="corner_refinement"           default="LINES" />

    <include file="$(find realsense2_camera)/launch/rs_rgbd.launch" />

    <node pkg="tf2_ros" type="static_transform_publisher" name="camera_to_hand" 
        args="-0.0187845171388 0.0629403083384 -0.0936029139899
                -0.5111042314 0.499085936412 0.50680045647 0.482533926313 
                mico_link_hand camera_link" />

    <!-- start ArUco -->
    <node name="aruco_tracker" pkg="aruco_ros" type="single">
        <remap from="/camera_info" to="/camera/color/camera_info" />
        <remap from="/image" to="/camera/color/image_rect_color" />
        <param name="image_is_rectified" value="true"/>
        <param name="marker_size"        value="$(arg marker_size)"/>
        <param name="marker_id"          value="$(arg marker_id)"/>
        <param name="reference_frame"    value="camera_link"/>
        <param name="camera_frame"       value="camera_color_optical_frame"/>
        <param name="marker_frame"       value="marker_frame" />
        <param name="corner_refinement"  value="$(arg corner_refinement)" />
    </node>

    <node name="rviz" pkg="rviz" type="rviz" output="screen" args="-d $(find arm_vs)/aruco.rviz"/>

    <node name="static_tf_node_desired_camera_frame" pkg="tf2_ros" type="static_transform_publisher" args="0 0.5 0 0.5 -0.5 0.5 0.5 marker_frame desired_cam_frame"/>

    <include file="$(find arm_vs)/launch/pbvs.launch">
        <arg name="control_gain"                                value="0.75" />
        <arg name="target_camera_frame"                         value="desired_cam_frame" />
        <arg name="current_camera_frame"                        value="camera_color_optical_frame" />
        <arg name="robot_base_frame"                            value="mico_link_base" />
    </include>
</launch>